app/src/main/java/com/cortado/mafkir/ui/adapter/MafkirContactListAdapter.kt

```diff
@@ -47,21 +48,28 @@ class MafkirContactListAdapter : RecyclerView.Adapter<MafkirContactListAdapter.V
         diffResult.dispatchUpdatesTo(this)
     }
 
-    class ViewHolder(private val binding: MafkircontactItemsBinding) :
+    class ViewHolder(private val context: Context, private val binding: MafkircontactItemsBinding) :
         RecyclerView.ViewHolder(binding.root) {
-        private val timeConverter = TimeConverter()
 
         fun bind(item: MafkirContact) {
             binding.mafkirContact = item
-            binding.timeConverter = timeConverter
-            if (System.currentTimeMillis() - item.lastInteractionMillis > item.interactionIntervalMillis) {
-                itemView.cardView.setCardBackgroundColor(Color.RED)
+            binding.intervalDisplayText =
+                "Every ${item.interval.interval} ${item.interval.type.text}"
+            if (intervalPassed(item)) {
+                itemView.cardView.setCardBackgroundColor(context.resources.getColor(R.color.colorError))
             }
             binding.clickListener = View.OnClickListener {
-                val navDirection = ListFragmentDirections.actionListFragmentToEditFragment(item)
-                val extras =  FragmentNavigatorExtras(binding.root.toTransitionGroup())
+                val navDirection = ListFragmentDirections.actionListFragmentToEditFragment(
+                    item.contact,
+                    true,
+                    item
+                )
+                val extras = FragmentNavigatorExtras(binding.root.toTransitionGroup())
                 it.findNavController().navigate(navDirection, extras)
             }
         }
+
+        // TODO: resolve interval-pass calculation, including notifying user
```

Comment:
why is the calculation not possible? just convert the interval to millis and do the same check as before

Reply:
dn